File: leet_code/python/9.palindrome-number.py
#
# @lc app=leetcode id=9 lang=python3
#
# [9] Palindrome Number
#
class Solution:
    def isPalindrome(self, x: int) -> bool:
        x = str(x)
        return x == x[::-1]
    def mathop(self, x):
        if x < 0: return False
        hi = 1
        while x // hi > 10:
            hi *= 10
        while hi > 10:
            a,x = divmod(x, hi)
            x,b = divmod(x, 10)
            hi = hi // 100
            if a != b:
                return False
        return True
if __name__ == "__main__":
    s = Solution()
    print(s.mathop(13221))

File: cpp/9.palindrome-number.cpp
/*
 * @lc app=leetcode id=9 lang=cpp
 *
 * [9] Palindrome Number
 */
#include<string>
using namespace std;

class Solution {
public:
    bool isPalindrome(int x) {
        if(x<0)
            return false;
        string x_str = to_string(x);
        int i = 0, j = x_str.length()-1;
        while(i<=j)
        {
            if(x_str[i] != x_str[j])
                return false;
            i++;j--;
        }
        return true;
    }
};

